bayesian_logistic_regression.py

```diff
@@ -0,0 +1,365 @@
+import math
+import os
+import torch
+import torch.distributions.constraints as constraints
+import pyro
+from pyro.optim import Adam, SGD
+from pyro.infer import SVI, Trace_ELBO, config_enumerate, TraceEnum_ELBO, Predictive
+import pyro.distributions as dist
+from pyro.infer.autoguide import AutoDelta
+from pyro import poutine    
+from pyro.poutine import trace, replay, block
+from functools import partial
+import numpy as np
+import scipy.stats
+from pyro.infer.autoguide import AutoDelta
+from collections import defaultdict
+import matplotlib
+from matplotlib import pyplot
+from pyro.infer import MCMC, NUTS
+import pandas as pd
+import pickle
+from pyro.infer.autoguide import AutoDiagonalNormal
+import inspect
+from bbbvi import relbo, Approximation
+
+PRINT_INTERMEDIATE_LATENT_VALUES = False
+PRINT_TRACES = False
+
+# this is for running the notebook in our testing framework
+smoke_test = ('CI' in os.environ)
+n_steps = 2 if smoke_test else 10000
+pyro.set_rng_seed(2)
+
+# enable validation (e.g. validate parameters of distributions)
+pyro.enable_validation(True)
```

Comment:
Do we want to set this globally forever?

Reply:
Thanks for taking a look at the logistic regression code as well. I've focused on the tutorial though and I haven't refactored the Bayesian Logistic Regression code yet. Will notify you when it's ready for review.

Reply:
sounds good.